ALEC-227: ALEC docs: update installation section

Bonrob2 opened the pull request:
Creating draft pull request, as the graphic in the deployment section still needs to be updated. Looking for review of text to ensure accuracy.

Bonrob2 committed:
ALEC-227: ALEC docs: update Installation section

ALEC-227: ALEC docs: update Installation section - going through the installation section of the documentation to make sure it is accurate for next ALEC release.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section
Went through each section and updated content based on new ALEC work. Still requires review and verification.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - streamlined ALEC installation docs to just go through a basic install.

Bonrob2 committed:
ALEC-227: ALEC DOCS: update installation section

ALEC-227: ALEC DOCS: update installation section - made additional changes to streamline the installation section.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - fixed some issues with local build, added information about ALEC installation.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - updated graphic with the basic deployment to remove HELM element. Copied original and png versions over.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - created new admin section and moved items to it.

Bonrob2 committed:
Merge branch 'develop' into jira/ALEC-227

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

Some tweaks to distributed installation section.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - updated the distributed deployment section, and some of the maintenance topics.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - updated the distributed deployment graphic

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
-= Distributed Install
+= Distributed Installation
 :imagesdir: ../assets/images
```
```suggestion

:imagesdir: ../assets/images
= Distributed Installation
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
 
-This installation guide describes a distributed install where we run _ALEC_ as a service on top of the _Sentinel_ runtime:
+A distributed installation runs ALEC as a service on top of https://docs.opennms.com/horizon/latest/deployment/sentinel/introduction.html[OpenNMSSentinel].
+OpenNMS Sentinel provides scalability for data processing.
```
```suggestion
A distributed deployment runs ALEC as a service on top of https://docs.opennms.com/horizon/latest/deployment/sentinel/introduction.html[Sentinel], an OpenNMS component that enables scalability for data processing.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
+A distributed installation runs ALEC as a service on top of https://docs.opennms.com/horizon/latest/deployment/sentinel/introduction.html[OpenNMSSentinel].
+OpenNMS Sentinel provides scalability for data processing.
 
 image::distributed_deployment.png[Distributed,700]
```
```suggestion
.Example distributed installation network architecture graph
image::distributed_deployment.png[Network architecture graph that displays the components of a distributed installation of ALEC, 700]
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
 == Prequisites
 
-In addition to an existing instance of OpenNMS Horizon 30.0.1 or greater, this guide assumes you have:
+In addition to an existing instance of OpenNMS Horizon 30.0.1 or greater, this guide assumes you have the following:
```
```suggestion
In addition to an existing instance of OpenNMS Horizon 30.0.1 or newer, this guide assumes that you have the following:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
-In addition to an existing instance of OpenNMS Horizon 30.0.1 or greater, this guide assumes you have:
+In addition to an existing instance of OpenNMS Horizon 30.0.1 or greater, this guide assumes you have the following:
 
 1. A Kafka broker (or cluster) running Kafka 0.11.x or greater.
```
```suggestion
* A Kafka broker (or cluster) running Kafka 0.11.x or greater.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
+3. Sentinel
+** Minions configured to use Kafka for the transport.
+** Minions ready to receive SNMP traps and Syslog messages.
+4. ZooKeeper 3.5.10+.
```
```suggestion
* One or more active Minions.
* OpenNMS Sentinel.
** Minions configured to use Kafka for the transport.
** Minions ready to receive SNMP traps and Syslog messages.
* ZooKeeper 3.5.10 or newer.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
+When installing the package repository, ensure that you do this on all of the systems that are running Sentinel and OpenNMS.
 
-== OpenNMS Configuration
+== OpenNMS configuration
```
```suggestion
== OpenNMS configuration

Complete the following steps through the OpenNMS Karaf shell.
They configure your OpenNMS instance in preparation for a distributed installation of ALEC.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
+== OpenNMS configuration
 
-=== Enable and configure the Kafka Producer (for topology & alarms)
+=== Enable and configure the Kafka Producer (for topology and alarms)
```
```suggestion
=== Enable and configure the Kafka Producer
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -1,36 +1,40 @@
-Configure the Kafka Producer to point to your Kafka broker:
 ```
 echo 'bootstrap.servers=127.0.0.1:9092' > "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.client.cfg"
 ```
````
```suggestion
[source, karaf]
----
echo 'bootstrap.servers=127.0.0.1:9092' > "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.client.cfg"
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -1,36 +1,40 @@
+=== Enable and configure the Kafka Producer (for topology and alarms)
+
+.Configure the Kafka Producer to point to your Kafka broker:
 
```
```suggestion
.Configure the Kafka Producer to point to your Kafka broker:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -1,36 +1,40 @@
+.Disable incremental alarm suppression:
 ```
 echo 'suppressIncrementalAlarms=false' >> "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.cfg"
 ```
````
```suggestion
[source, karaf]
----
echo 'suppressIncrementalAlarms=false' >> "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.cfg"
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
+
 ```
 echo 'org.opennms.netmgt.eventd.sink.enable=true' > "$OPENNMS_HOME/etc/opennms.properties.d/event-sink.properties"
 ```
````
```suggestion
[source, karaf]
----
echo 'org.opennms.netmgt.eventd.sink.enable=true' > "$OPENNMS_HOME/etc/opennms.properties.d/event-sink.properties"
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
 == Install ALEC in OpenNMS
 
-On the OpenNMS server, install the package:
+.On the OpenNMS server, install the package:
+
```
```suggestion
.Install the ALEC plugin package on your OpenNMS server:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
-On the OpenNMS server, install the package:
+.On the OpenNMS server, install the package:
+
 ```
 yum -y install opennms-alec-plugin
 ```
````
```suggestion
[source, karaf]
yum -y install opennms-alec-plugin
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
 ```
 
-Configure the feature set to be installed when OpenNMS starts:
+.Configure the feature set to be installed when OpenNMS starts:
+
````
```suggestion
.Configure the feature set to be installed when OpenNMS starts:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
-Configure the feature set to be installed when OpenNMS starts:
+.Configure the feature set to be installed when OpenNMS starts:
+
 ```
 echo 'alec-opennms-distributed wait-for-kar=opennms-alec-plugin' > "$OPENNMS_HOME/etc/featuresBoot.d/alec.boot"
 ```
````
```suggestion
[source, karaf]
----
echo 'alec-opennms-distributed wait-for-kar=opennms-alec-plugin' > "$OPENNMS_HOME/etc/featuresBoot.d/alec.boot"
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
 == Install ALEC in Sentinel
 
-On the Sentinel systems, install the package:
+.On the Sentinel systems, install the package:
+
```
```suggestion
.Install the ALEC plugin package on your Sentinel systems:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
-On the Sentinel systems, install the package:
+.On the Sentinel systems, install the package:
+
 ```
 yum -y install sentinel-alec-plugin
 ```
````
```suggestion
[source, karaf]
yum -y install sentinel-alec-plugin
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
 ```
 
-Configure the Kafka Stream (consumer):
+.Configure the Kafka Stream (consumer):
+
````
```suggestion
.Configure the Kafka Stream (consumer):
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -39,33 +43,39 @@ Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOM
-Configure the Kafka Stream (consumer):
+.Configure the Kafka Stream (consumer):
+
 ```
 echo 'bootstrap.servers=127.0.0.1:9092
 commit.interval.ms=5000' > "$SENTINEL_HOME/etc/org.opennms.alec.datasource.opennms.kafka.streams.cfg"
````
This can be made into a source block using the syntax from my previous suggestions. I'd make the change, but I can't suggest edits for the final line of the snippet.

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -76,14 +86,14 @@ Configure the Kafka producer:
 echo 'bootstrap.servers=127.0.0.1:9092' > "$SENTINEL_HOME/etc/org.opennms.alec.datasource.opennms.kafka.producer.cfg"
 ```
````
Same as above, but with the first line this time.

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -76,14 +86,14 @@ Configure the Kafka producer:
 
-Configure ZooKeeper for distributed coordination:
+.Configure ZooKeeper for distributed coordination:
 ```
 echo 'connectString=127.0.0.1:2181' > "$SENTINEL_HOME/etc/org.opennms.features.distributed.coordination.zookeeper.cfg"
 ```
````
```suggestion
[source, karaf]
----
echo 'connectString=127.0.0.1:2181' > "$SENTINEL_HOME/etc/org.opennms.features.distributed.coordination.zookeeper.cfg"
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -76,14 +86,14 @@ Configure the Kafka producer:
 
-Configure the feature set to be installed when Sentinel starts:
+.Configure the feature set to be installed when Sentinel starts:
 ```
 echo 'sentinel-core
 sentinel-coordination-zookeeper
````
Same as the previous code block comments.

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
 Ensure that the required Kafka topics are created.
 See xref:datasources:kafka.adoc#_topics[topics] for details.
```
```suggestion
Ensure that the required Kafka topics are created (see xref:datasources:kafka.adoc#_topics[topics]).
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
-Run the `health:check` commands in both OpenNMS and Sentinel.
+Run the `health:check` command in both OpenNMS and Sentinel.
+
+From a Sentinel Karaf shell, enumerate the available graphs using the following:
```
```suggestion
From a Sentinel Karaf shell, enumerate the available graphs using the following command:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
+From a Sentinel Karaf shell, enumerate the available graphs using the following:
 
-From a Sentinel Karaf shell, enumerate the available graphs using:
 ```
 opennms-alec:list-graphs
 ```
````
```suggestion
[source, karaf]
opennms-alec:list-graphs
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
 opennms-alec:list-graphs
 ```
 
 At this point we expect the command output a single graph called "dbscan":
````
```suggestion
At this point we expect the command to output a single graph called "dbscan":
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
 
 At this point we expect the command output a single graph called "dbscan":
+
 ```
````
```suggestion
[source, karaf]
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
-admin@opennms> opennms-alec:list-graphs 
+admin@opennms> opennms-alec:list-graphs
 dbscan: 0 situations on 524 vertices and 4 edges.
 ```
````
```suggestion
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -97,19 +107,19 @@ Restart OpenNMS and Sentinel.
-It's okay if there are no situations, vertices or edges on the graph at this point.
-
-Continue to xref:verifying.adoc[Verifying your install] for more details.
+It's okay if there are no situations, vertices, or edges on the graph at this point.
```
```suggestion
NOTE: There may not be any situations, vertices, or edges on the graph at this point.
```

emarsh1995 commented on docs/modules/admin/pages/overview.adoc:
```diff
@@ -0,0 +1,4 @@
+= Overview
+:imagesdir: ../assets/images
```
```suggestion

:imagesdir: ../assets/images
= Overview
```

emarsh1995 commented on docs/modules/admin/pages/overview.adoc:
```diff
@@ -0,0 +1,4 @@
+= Overview
+:imagesdir: ../assets/images
+
+This section describes maintenance tasks and more advanced procedures, including distributed deployment and how to upgrade ALEC.
```
```suggestion
This section describes maintenance tasks and advanced procedures, including distributed deployment and how to upgrade ALEC.
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -1,9 +1,9 @@
-= Upgrading
+= Upgrade ALEC
 :imagesdir: ../assets/images
```
```suggestion

:imagesdir: ../assets/images
= Upgrade ALEC
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -1,9 +1,9 @@
 :imagesdir: ../assets/images
 
-General upgrading instructions for ALEC.
+This section includes instructions to upgrade ALEC on OpenNMS and Sentinel.
```
```suggestion
This section describes how to upgrade ALEC on OpenNMS and Sentinel.
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -1,9 +1,9 @@
+This section includes instructions to upgrade ALEC on OpenNMS and Sentinel.
 
-== Upgrading ALEC on OpenNMS (monolithic & distributed install)
+== Upgrade ALEC on OpenNMS (basic and distributed deployments)
```
```suggestion
== Upgrade ALEC on OpenNMS

NOTE: The following instructions apply to both basic and distributed ALEC deployments.
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -1,9 +1,9 @@
-== Upgrading ALEC on OpenNMS (monolithic & distributed install)
+== Upgrade ALEC on OpenNMS (basic and distributed deployments)
 
 Stop OpenNMS:
```
I'd change all of these to code blocks as well, just for consistency.

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
````diff
@@ -25,10 +25,10 @@ Restart OpenNMS:
 ```
 
-== Upgrading ALEC on Sentinel (distributed install)
+== Upgrade ALEC on Sentinel (distributed deployment)
````
```suggestion
== Upgrade ALEC on Sentinel

NOTE: The following instructions apply to only a distributed ALEC installation.
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -25,10 +25,10 @@ Restart OpenNMS:
 
-Here we assume that ALEC is already up and running on 1+ Sentinel instances.
-For each of these, proceed as follows:
+Here we assume that ALEC is already up and running on one or more Sentinel instances.
+For each Sentinel instance, do the following:
```
```suggestion
For this procedure, we assume that ALEC is already installed and running on one or more Sentinel instances in your network.
Follow the steps below for each Sentinel instance:
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -25,10 +25,10 @@ Restart OpenNMS:
+Here we assume that ALEC is already up and running on one or more Sentinel instances.
+For each Sentinel instance, do the following:
 
 Stop Sentinel:
```
Same as previous code block suggestions.

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+= Maintenance Tasks
+:imagesdir: ../assets/images
```
```suggestion

:imagesdir: ../assets/images
= Maintenance Tasks
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+= Maintenance Tasks
+:imagesdir: ../assets/images
+
+The procedures in this section describe maintenance tasks you may want to perform.
```
```suggestion
The procedures in this section describe maintenance tasks that you may want to perform.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+
+To verify that syslog messages are being properly converted to alarms and show that ALEC can create a situation from multiple alarms, do the following:
+
+.Provision a node for the `localhost` in OpenNMS:
+
+```
+$OPENNMS_HOME/bin/provision.pl requisition add alec
+$OPENNMS_HOME/bin/provision.pl node add alec localhost localhost
+$OPENNMS_HOME/bin/provision.pl interface add alec localhost 127.0.0.1
+$OPENNMS_HOME/bin/provision.pl requisition import alec
+```
````
```suggestion
.Provision a node for the `localhost` in OpenNMS:
[source]
----
$OPENNMS_HOME/bin/provision.pl requisition add alec
$OPENNMS_HOME/bin/provision.pl node add alec localhost localhost
$OPENNMS_HOME/bin/provision.pl interface add alec localhost 127.0.0.1
$OPENNMS_HOME/bin/provision.pl requisition import alec
----
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+
+IMPORTANT: This assumes that the localhost is running an SNMP agent.
+
+.Set the `ifDescr` for an interface on the node we just provisioned.
+
+```
+export IFDESCR="eth0"
+```
````
```suggestion
.Set the `ifDescr` for an interface on the node we just provisioned.
[source]
export IFDESCR="eth0"
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+```
+export IFDESCR="eth0"
+```
+.Trigger a Syslog message:
+
+```
+echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %ETHPORT-5-IF_DOWN_LINK_FAILURE: Interface $IFDESCR is down (Link failure)" | nc -v -u 127.0.0.1 10514
+```
````
```suggestion

.Trigger a Syslog message:
[source]
----
echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %ETHPORT-5-IF_DOWN_LINK_FAILURE: Interface $IFDESCR is down (Link failure)" | nc -v -u 127.0.0.1 10514
----

```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+```
+echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %ETHPORT-5-IF_DOWN_LINK_FAILURE: Interface $IFDESCR is down (Link failure)" | nc -v -u 127.0.0.1 10514
+```
+We expect to see a new "interface down" alarm attached to the node we just provisioned:
````
```suggestion
You should see a new "interface down" alarm attached to the node that you just provisioned:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+image::verify_ifdown_example.png[Interface down alarm,600]
+
+TIP: If the Syslog message is not being recognized and bring formatted to a proper event and alarm, ensure that the `opennms-alec-plugin` feature is running in OpenNMS and that the related bundles are active.
```
```suggestion
TIP: If the Syslog message is not being recognized and formatted to a proper event and alarm, ensure that the `opennms-alec-plugin` feature is running in OpenNMS and that the related bundles are active.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+
+TIP: If the Syslog message is not being recognized and bring formatted to a proper event and alarm, ensure that the `opennms-alec-plugin` feature is running in OpenNMS and that the related bundles are active.
+
+.Next, trigger a second alarm against this same interface:
+
+```
+echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %PKT_INFRA-LINEPROTO-5-UPDOWN: Line protocol on Interface $IFDESCR, changed state to Down" | nc -v -u 127.0.0.1 10514
+```
````
```suggestion
.Trigger a second alarm against this same interface:
[source]
----
echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %PKT_INFRA-LINEPROTO-5-UPDOWN: Line protocol on Interface $IFDESCR, changed state to Down" | nc -v -u 127.0.0.1 10514
----

```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+```
+echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %PKT_INFRA-LINEPROTO-5-UPDOWN: Line protocol on Interface $IFDESCR, changed state to Down" | nc -v -u 127.0.0.1 10514
+```
+If both of these alarms are triggered together in a small enough window of time (within one minute for the purposes of this example), ALEC should automatically create a situation:
````
```suggestion
If both of these alarms are triggered within a brief window of time (within one minute, for the purposes of this example), ALEC should automatically create a situation:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+image::verify_situation_example.png[Situation,600]
+
+== View the ALEC graph
```
```suggestion
== View network inventory graph
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+== View the ALEC graph
+
+At any time, you can export the ALEC graph to view the inventory and alarms in context.
```
```suggestion
At any time, you can export the network inventory graph that ALEC maintains to view your network's inventory, alarms, and situations in context.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+
+```
+opennms-alec:export-graph dbscan /tmp/dbscan.graph.xml
+```
````
```suggestion
.Export the ALEC graph from a Karaf shell:
[source, karaf]
opennms-alec:export-graph dbscan /tmp/dbscan.graph.xml
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+
+NOTE: If ALEC is running on Sentinel, make sure to run this from the Sentinel shell.
+
+.Import the graph into OpenNMS:
+
+```
+curl -X POST -H "Content-Type: application/xml" -u admin:admin -d@/tmp/dbscan.graph.xml 'http://localhost:8980/opennms/rest/graphml/alec'
+```
````
```suggestion
.Import the graph into OpenNMS:
[source]
----
curl -X POST -H "Content-Type: application/xml" -u admin:admin -d@/tmp/dbscan.graph.xml 'http://localhost:8980/opennms/rest/graphml/alec'
----
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
````diff
@@ -0,0 +1,71 @@
+curl -X POST -H "Content-Type: application/xml" -u admin:admin -d@/tmp/dbscan.graph.xml 'http://localhost:8980/opennms/rest/graphml/alec'
+```
+
+NOTE: If you get an error that a `"Graph with name alec already exists"`, you can delete the previous graph using: `curl -X DELETE -u admin:admin 'http://localhost:8980/opennms/rest/graphml/alec'` and try again.
````
```suggestion
NOTE: If a "Graph with name alec already exists" error is displayed, delete the previous graph using the following command: `curl -X DELETE -u admin:admin 'http://localhost:8980/opennms/rest/graphml/alec'`.
After the graph is deleted, you can try to create it again by following the steps above.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+.Next, navigate to the topology map, and view the graph named "alec".
+
+The resulting graph from the previous example looks like the following:
```
```suggestion
After the graph is created, you can navigate to the OpenNMS topology map to view it.
The graph resulting from this example is displayed as below:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+The resulting graph from the previous example looks like the following:
+
+image::verify_topology_example.png[Topology,600]
```
```suggestion
image::verify_topology_example.png[Network topology map, 600]
```

emarsh1995 commented on docs/modules/install/pages/basic_install.adoc:
```diff
@@ -0,0 +1,47 @@
+= Basic Installation
+:imagesdir: ../assets/images
```
```suggestion

:imagesdir: ../assets/images
= Basic Installation
```

emarsh1995 commented on docs/modules/install/pages/basic_install.adoc:
```diff
@@ -0,0 +1,47 @@
+= Basic Installation
+:imagesdir: ../assets/images
+
+This section describes a basic installation where we run ALEC as another service within the OpenNMS core:
```
```suggestion
This section describes a basic installation, where ALEC runs as another service within the OpenNMS core instance:
```

emarsh1995 commented on docs/modules/install/pages/basic_install.adoc:
```diff
@@ -0,0 +1,47 @@
+
+This section describes a basic installation where we run ALEC as another service within the OpenNMS core:
+
+image::basic_deployment.png[basic ALEC installation,400]
```
```suggestion
.Example basic installation network architecture graph
image::basic_deployment.png[Network architecture graph that displays a basic installation of ALEC, 400]
```

emarsh1995 commented on docs/modules/install/pages/basic_install.adoc:
```diff
@@ -0,0 +1,47 @@
+
+Make sure you complete the xref:pre_install.adoc[pre-installation tasks] before proceeding with the basic installation.
+
+== Install ALEC
```
I'd like to figure out how to use the tabbed blocks for these instructions, like we do in the core docs.

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+
+= Installation Overview
+
+There are two options for ALEC deployment: basic and distributed.
```
```suggestion
There are two options available for ALEC deployment: basic and distributed installations.
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+
+A xref:basic_install.adoc[basic deployment], which installs ALEC in the OpenNMS core as an additional service, is easy to set up and lets you evaluate or test ALEC with your existing environment.
+
+A xref:admin:distributed_install.adoc[distributed deployment], which installs ALEC externally on an OpenNMS Sentinel server(s), requires a more involved setup that we recommend for a production environment where you have load, scaling, and resiliency requirements.
```
```suggestion
A xref:admin:distributed_install.adoc[distributed deployment], which installs ALEC externally on one or more OpenNMS Sentinel servers, requires a more involved setup that we recommend for a production environment where you have load, scaling, and resiliency requirements.
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+
+A xref:admin:distributed_install.adoc[distributed deployment], which installs ALEC externally on an OpenNMS Sentinel server(s), requires a more involved setup that we recommend for a production environment where you have load, scaling, and resiliency requirements.
+
+.Basic vs. Distributed deployment pros and cons
```
```suggestion
.Basic vs. distributed deployment pros and cons
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+| Additional load on your OpenNMS instance.
+
+| Distributed
+| 1. Take load off of the OpenNMS core.
```
```suggestion
| 1. Removes load from the OpenNMS core.
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+
+| Distributed
+| 1. Take load off of the OpenNMS core.
+
+2. Fault tolerant setup (ALEC runs in high availability).
```
```suggestion
2. Fault tolerant setup (ALEC runs in high availability).
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+
+ALEC requires the following:
+
+* Horizon 30.0.1+ or Meridian 2022+ up and running.
+(See link:https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS.)
```
```suggestion
* Horizon 30.0.1+ or Meridian 2022+ (see link:https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS).
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+
+* Horizon 30.0.1+ or Meridian 2022+ up and running.
+(See link:https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS.)
+* https://www.elastic.co/products/elasticsearch[Elasticsearch] 6.7.0–7.6.2 to maintain event and alarm history for training purposes.
+(See https://docs.opennms.com/horizon/31/operation/deep-dive/elasticsearch/introduction.html[Elasticsearch integration] for more information.)
```
```suggestion
* https://www.elastic.co/products/elasticsearch[Elasticsearch] 6.7.0–7.6.2 to maintain event and alarm history for training purposes (see https://docs.opennms.com/horizon/latest/operation/deep-dive/elasticsearch/introduction.html[Elasticsearch integration]).
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,40 @@
+* https://www.elastic.co/products/elasticsearch[Elasticsearch] 6.7.0–7.6.2 to maintain event and alarm history for training purposes.
+(See https://docs.opennms.com/horizon/31/operation/deep-dive/elasticsearch/introduction.html[Elasticsearch integration] for more information.)
+
+== Before you begin
```
```suggestion
== Before you begin

```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,127 @@
+= Pre-Installation Tasks
+:imagesdir: ../assets/images
```
```suggestion

:imagesdir: ../assets/images
= Pre-Installation Tasks
```

emarsh1995 reviewed (changes requested):
I got most of the way through, but I'll have to finish reviewing this tomorrow. :)

Bonrob2 committed:
Apply suggestions from code review

Co-authored-by: Emily <107147744+emarsh1995@users.noreply.github.com>

Bonrob2 committed:
Apply suggestions from code review

Co-authored-by: Emily <107147744+emarsh1995@users.noreply.github.com>

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - some updates based on reviewer feedback.

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+:imagesdir: ../assets/images
+= Distributed Installation
+
+A distributed deployment runs ALEC as a service on top of https://docs.opennms.com/horizon/latest/deployment/sentinel/introduction.html[Sentinel], an OpenNMS component that enables scalability for data processing.
```
```suggestion
A distributed installation runs ALEC as a service on top of https://docs.opennms.com/horizon/latest/deployment/sentinel/introduction.html[Sentinel], an OpenNMS component that enables scalability for data processing.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+
+== Before you begin
+
+Make sure you complete the xref:install:pre_install.adoc[pre-installation tasks] before proceeding with the basic installation.
```
```suggestion
You must complete the xref:install:pre_install.adoc[pre-installation tasks] before proceeding with the installation.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+
+Make sure you complete the xref:install:pre_install.adoc[pre-installation tasks] before proceeding with the basic installation.
+
+When installing the package repository, ensure that you do this on all of the systems that are running Sentinel and OpenNMS.
```
```suggestion
Make sure that you install the package repository on all systems that are running Sentinel and OpenNMS.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+
+== OpenNMS configuration
+
+Complete the following steps through the OpenNMS Karaf shell.
```
```suggestion
Complete the following steps in the OpenNMS Karaf shell.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+
+Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOME/etc/org.apache.karaf.features.cfg`.
+
+=== Enable consumption of events from Kafka
```
```suggestion
=== Enable consumption of Kafka events
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -0,0 +1,129 @@
+```
+
+IMPORTANT: We assume that you are already using the Kafka strategy for the Sink API.
+See <<prequisites>>.
````
```suggestion
IMPORTANT: This procedure assumes that you are already using the Kafka strategy for the Sink API (see <<prequisites>>).
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -0,0 +1,129 @@
+echo 'bootstrap.servers=127.0.0.1:9092' > "$SENTINEL_HOME/etc/org.opennms.alec.datasource.opennms.kafka.producer.cfg"
+```
+
+TIP: The consumer reads alarms and inventory, while the producer sends events.
````
```suggestion
TIP: Kafka Stream reads alarms and inventory, while the Kafka producer sends events.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -0,0 +1,129 @@
+alec-sentinel-distributed wait-for-kar=opennms-alec-plugin' >> "$SENTINEL_HOME/etc/featuresBoot.d/alec.boot"
+```
+
+== Verify
````
```suggestion
== Verify your installation
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+----
+admin@opennms> opennms-alec:list-graphs
+dbscan: 0 situations on 524 vertices and 4 edges.
+----
```
```suggestion
Follow the steps below to verify your distributed ALEC installation:

. Restart your OpenNMS instance and any Sentinel instances.
. Make sure that all required Kafka topics have been created (see xref:datasources:kafka.adoc#_topics[Kafka topics]).
. Run the `health:check` command in an OpenNMS Karaf shell to check the status of your OpenNMS instance.
. Run the `health:check` command in a Sentinel Karaf shell to check the status of the Sentinel instance.
** Repeat this step for each Sentinel instance in your environment.
. Run the `opennms-alec:list-graphs` in a Sentinel Karaf shell to enumerate all available graphs.

At this point, the health check commands should return with no issues, and one graph named "dbscan" should have been enumerated:

[source, karaf]
----
admin@opennms> opennms-alec:list-graphs
dbscan: 0 situations on 524 vertices and 4 edges.
----
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,129 @@
+dbscan: 0 situations on 524 vertices and 4 edges.
+----
+
+NOTE: There may not be any situations, vertices, or edges on the graph at this point.
```
```suggestion
NOTE: Your graph may not have any situations, vertices, or edges at this point.
This is normal during initial setup.
```

emarsh1995 commented on docs/modules/admin/pages/overview.adoc:
```diff
@@ -0,0 +1,5 @@
+:imagesdir: ../assets/images
+= Overview
+
+This section describes maintenance tasks and advanced procedures, including distributed deployment and how to upgrade ALEC.
```
```suggestion
This section describes maintenance tasks and advanced procedures, including distributed installation instructions and how to upgrade ALEC.
```

emarsh1995 commented on docs/modules/admin/pages/upgrading.adoc:
```diff
@@ -1,9 +1,12 @@
+== Upgrade ALEC on OpenNMS
 
-== Upgrading ALEC on OpenNMS (monolithic & distributed install)
+NOTE: The following instructions apply to both basic and distributed ALEC deployments.
```
```suggestion
NOTE: The following instructions apply to both basic and distributed ALEC installations.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+:imagesdir: ../assets/images
+= Maintenance Tasks
+
+The procedures in this section describe maintenance tasks that you may want to perform.
```
```suggestion
This section describes maintenance tasks that you may want to perform.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+== Test syslog message handling
+
+To verify that syslog messages are being properly converted to alarms and show that ALEC can create a situation from multiple alarms, do the following:
```
```suggestion
Follow the steps below to verify that syslog messages are being properly converted to alarms, and that ALEC can create a situation:
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -0,0 +1,129 @@
+echo 'suppressIncrementalAlarms=false' >> "$OPENNMS_HOME/etc/org.opennms.features.kafka.producer.cfg"
+```
+
+Add `opennms-kafka-producer` to the `featuresBoot` property in the `$OPENNMS_HOME/etc/org.apache.karaf.features.cfg`.
````
```suggestion
Open `$OPENNMS_HOME/etc/org.apache.karaf.features.cfg` in a text editor and add `opennms-kafka-producer` to the `featuresBoot` property.
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -0,0 +1,129 @@
+yum -y install sentinel-alec-plugin
+```
+
+.Configure the Kafka Stream (consumer):
````
```suggestion
.Configure Kafka Stream (consumer):
```

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
````diff
@@ -0,0 +1,129 @@
+commit.interval.ms=5000' > "$SENTINEL_HOME/etc/org.opennms.alec.datasource.opennms.kafka.streams.cfg"
+```
+
+Configure the Kafka producer:
````
```suggestion
.Configure the Kafka producer:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+IMPORTANT: This assumes that the localhost is running an SNMP agent.
+
+.Set the `ifDescr` for an interface on the node we just provisioned.
```
```suggestion
.Set `ifDescr` for an interface on the provisioned node:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+[source]
+export IFDESCR="eth0"
+
+.Trigger a Syslog message:
```
```suggestion
.Trigger a syslog message:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %ETHPORT-5-IF_DOWN_LINK_FAILURE: Interface $IFDESCR is down (Link failure)" | nc -v -u 127.0.0.1 10514
+----
+
+You should see a new "interface down" alarm attached to the node that you just provisioned:
```
```suggestion
You should see a new "interface down" alarm associated with the node that you just provisioned:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+image::verify_ifdown_example.png[Interface down alarm,600]
+
+TIP: If the Syslog message is not being recognized and formatted to a proper event and alarm, ensure that the `opennms-alec-plugin` feature is running in OpenNMS and that the related bundles are active.
```
```suggestion
TIP: If the syslog message is not being recognized and formatted as a proper event and alarm, ensure that the `opennms-alec-plugin` feature is running in OpenNMS and that the related bundles are active.
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+echo "<189>: $(date +"%Y %b %d %H:%m:%S %Z"): %PKT_INFRA-LINEPROTO-5-UPDOWN: Line protocol on Interface $IFDESCR, changed state to Down" | nc -v -u 127.0.0.1 10514
+----
+
+If both of these alarms are triggered within a brief window of time (within one minute, for the purposes of this example), ALEC should automatically create a situation:
```
```suggestion
If both alarms are triggered within a brief window of time (within one minute, for the purposes of this example), ALEC should automatically create a situation:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+
+At any time, you can export the network inventory graph that ALEC maintains to view your network's inventory, alarms, and situations in context.
+
+.Export the ALEC graph from a Karaf shell:
```
```suggestion
.Export the ALEC graph:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+[source, karaf]
+opennms-alec:export-graph dbscan /tmp/dbscan.graph.xml
+
+NOTE: If ALEC is running on Sentinel, make sure to run this from the Sentinel shell.
```
```suggestion
NOTE: If ALEC is running on Sentinel, make sure to run this command from the Sentinel Karaf shell.
```

emarsh1995 commented on docs/modules/install/pages/basic_install.adoc:
```diff
@@ -0,0 +1,49 @@
+
+== Before you begin
+
+Make sure you complete the xref:pre_install.adoc[pre-installation tasks] before proceeding with the basic installation.
```
```suggestion
You must complete the xref:install:pre_install.adoc[pre-installation tasks] before proceeding with the installation.
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,38 @@
+
+= Installation Overview
+
+There are two options available for ALEC deployment: basic and distributed installations.
+
+A xref:basic_install.adoc[basic deployment], which installs ALEC in the OpenNMS core as an additional service, is easy to set up and lets you evaluate or test ALEC with your existing environment.
+
+A xref:admin:distributed_install.adoc[distributed deployment], which installs ALEC externally on one or more OpenNMS Sentinel servers, requires a more involved setup that we recommend for a production environment where you have load, scaling, and resiliency requirements.
```
```suggestion
There are two options available for ALEC deployment: basic and distributed installations.
A xref:basic_install.adoc[basic deployment], which installs ALEC in the OpenNMS core as an additional service, is easy to set up and lets you evaluate or test ALEC with your existing environment.
A xref:admin:distributed_install.adoc[distributed deployment], which installs ALEC externally on one or more OpenNMS Sentinel servers, requires a more involved setup that we recommend for a production environment where you have load, scaling, and resiliency requirements.
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,38 @@
+
+A xref:admin:distributed_install.adoc[distributed deployment], which installs ALEC externally on one or more OpenNMS Sentinel servers, requires a more involved setup that we recommend for a production environment where you have load, scaling, and resiliency requirements.
+
+.Basic vs. distributed deployment pros and cons
```
```suggestion
.Basic versus distributed installation
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,38 @@
+[options="header"]
+[cols="1,2,2"]
+|===
+| Deployment
```
```suggestion
| Installation
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,38 @@
+
+ALEC requires the following:
+
+* Horizon 30.0.1+ or Meridian 2022+ (see link:https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS).
```
```suggestion
* Horizon 30.0.1+ or Meridian 2022+ (see https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS).
```

emarsh1995 commented on docs/modules/install/pages/introduction.adoc:
```diff
@@ -0,0 +1,38 @@
+
+== Before you begin
+
+You must complete the xref:pre_install.adoc[pre-installation tasks] before doing a basic or distributed deployment.
```
I think this section is redundant, since we have the same section in both of the installation pages.

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+
+This section describes the configuration tasks you must complete before installing ALEC.
+
+== OpenNMS configuration
```
I think we should use the tabbed block here too, if we can. :)

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+
+== OpenNMS configuration
+
+=== Set $OPENNMS_HOME variable
```
```suggestion
=== Set `$OPENNMS_HOME` variable
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+export OPENNMS_HOME=/usr/share/opennms
+```
+
+=== Validate your OpenNMS deployment
````
```suggestion
=== Verify your OpenNMS installation
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+
+```
+health:check
+```
````
```suggestion
Run the `health:check` command in an OpenNMS Karaf shell to verify the status of your OpenNMS instance.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+health:check
+```
+If the health check returns an `everything is awesome` message, continue with the next pre-installation task.
+An `Oh no, something is wrong` message indicates a problem you must address.
````
```suggestion
If the health check returns "Everything is awesome," continue to the next pre-installation task.
An "Oh no, something is wrong" error message indicates a problem that you must address before moving forward.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+Connecting to ElasticSearch ReST API (Flows)   [ Timeout  ] => Health Check did not finish within 5000 ms
+
+=> Oh no, something is wrong
+```
````
```suggestion
[source, karaf]
----
admin@opennms> health:check
Verifying the health of the container

Verifying installed bundles                    [ Success  ]
Connecting to ElasticSearch ReST API (Flows)   [ Timeout  ] => Health Check did not finish within 5000 ms

=> Oh no, something is wrong
----

```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+
+=> Oh no, something is wrong
+```
+OpenNMS deployments install the `opennms-flows` feature by default, but may not point to the Elasticsearch server.
````
```suggestion
OpenNMS deployments install the `opennms-flows` feature set by default, but may not point to the Elasticsearch server.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+```
+OpenNMS deployments install the `opennms-flows` feature by default, but may not point to the Elasticsearch server.
+
+To fix this issue, you can either uninstall the feature (using `feature:uninstall opennms-flows`), or point the feature to an Elasticsearch instance:
````
```suggestion
To fix this issue, you can either uninstall the feature (using the `feature:uninstall opennms-flows` command), or point the feature to an Elasticsearch instance:
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+config:edit org.opennms.features.flows.persistence.elastic
+config:property-set elasticUrl http://elastic:9200
+config:update
+```
````
```suggestion
[source, karaf]
----
config:edit org.opennms.features.flows.persistence.elastic
config:property-set elasticUrl http://elastic:9200
config:update
----

```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+```
+TIP: Replace the sample URL with one that points to your Elasticsearch instance.
+
+Run the health check again to confirm that `everything is awesome`.
````
```suggestion
Run the `health:check` command again and, if the "Everything is awesome" message is displayed, continue to the next tasks.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+
+=== Enable alarm history storage for Elasticsearch
+
+When using ALEC, we strongly recommend that you leverage our integration for maintaining alarm history in Elasticsearch.
```
```suggestion
We strongly recommended that, when using ALEC, you leverage the Elasticsearch integration.
This integration allows OpenNMS to maintain historical alarm information in Elasticsearch.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+When using ALEC, we strongly recommend that you leverage our integration for maintaining alarm history in Elasticsearch.
+ALEC can use this data for analysis and to help train the correlation engine.
+
+.From the OpenNMS Karaf shell, run the following, replacing the sample URL with one that points to your Elasticsearch instance:
```
```suggestion
To set up the integration, run the following commands in an OpenNMS Karaf shell:

.Point your OpenNMS instance to an Elasticsearch instance:
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+config:edit org.opennms.features.alarms.history.elastic
+config:property-set elasticUrl http://elastic:9200
+config:update
+```
````
```suggestion
[source, karaf]
----
config:edit org.opennms.features.alarms.history.elastic
config:property-set elasticUrl http://elastic:9200
config:update
----

IMPORTANT: Replace the sample URL (`http://elastic:9200`) with one that points to your Elasticsearch instance.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+config:property-set elasticUrl http://elastic:9200
+config:update
+```
+TIP: Replace the sample URL with one that points to your Elasticsearch instance.
````
```suggestion
NOTE: Replace the sample URL (`http://elastic:9200`) with one that points to your Elasticsearch instance.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+
+```
+feature:install opennms-alarm-history-elastic
+```
````
```suggestion
.Install the feature:
[source, karaf]
feature:install opennms-alarm-history-elastic
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+
+.Run the `health:check`
+
+If everything is configured properly, you should receive an `everything is awesome` message.
```
```suggestion
After the feature is installed, run the `health:check` command.
If everything is configured properly, an "Everything is awesome" message is displayed.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+
+If everything is configured properly, you should receive an `everything is awesome` message.
+
+=== Enable Syslogd
```
```suggestion
=== Enable syslogd
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+
+=== Enable Syslogd
+
+We recommend that you enable Syslogd to take advantage of the Syslog patterns and event definitions that ALEC provides:
```
```suggestion
We recommend that you enable syslogd to take advantage of the syslog patterns and event definitions that ALEC provides.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+
+We recommend that you enable Syslogd to take advantage of the Syslog patterns and event definitions that ALEC provides:
+
+.Edit `$OPENNMS_HOME/etc/service-configuration.xml`
+and change the following:
+
+```
+<service enabled="false">
+    <name>OpenNMS:Name=Syslogd</name>
+```
+
+to:
+
+```
+<service>
+    <name>OpenNMS:Name=Syslogd</name>
+```
````
```suggestion
First, open `$OPENNMS_HOME/etc/service-configuration.xml` in a text editor and change the following code:

[source]
----
<service enabled="false">
    <name>OpenNMS:Name=Syslogd</name>
----

to:

[source]
----
<service>
    <name>OpenNMS:Name=Syslogd</name>
----
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
````diff
@@ -0,0 +1,128 @@
+```
+
+This change will come into effect the next time OpenNMS restarts.
+Do not restart OpenNMS at this point, as you will restart it as part of the ALEC installation.
````
```suggestion
Do not restart OpenNMS at this point; you will restart it as part of the ALEC installation procedure.
```

emarsh1995 commented on docs/modules/install/pages/pre_install.adoc:
```diff
@@ -0,0 +1,128 @@
+This change will come into effect the next time OpenNMS restarts.
+Do not restart OpenNMS at this point, as you will restart it as part of the ALEC installation.
+
+== Set up ALEC/OpenNMS package repositories
```
```suggestion
== Set up package repositories
```

emarsh1995 commented on docs/modules/install/pages/requirements.adoc:
```diff
@@ -1,8 +1,7 @@
 = Requirements
 
 Before installing ALEC, we assume that you already have an instance of OpenNMS Horizon 30.0.1 or greater up and running.
```
```suggestion
Before installing ALEC, we assume that you already have an instance of OpenNMS Horizon 30.0.1 or greater up and running (see https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS).
```

emarsh1995 commented on docs/modules/install/pages/requirements.adoc:
```diff
@@ -1,8 +1,7 @@
 Before installing ALEC, we assume that you already have an instance of OpenNMS Horizon 30.0.1 or greater up and running.
-
-TIP: See link:https://www.opennms.org/en/install[OpenNMS Installation] for help on getting started with OpenNMS.
+(See link:https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS.)
```
```suggestion
```

emarsh1995 commented on docs/modules/install/pages/requirements.adoc:
```diff
@@ -1,8 +1,7 @@
-TIP: See link:https://www.opennms.org/en/install[OpenNMS Installation] for help on getting started with OpenNMS.
+(See link:https://docs.opennms.com/horizon/latest/deployment/core/introduction.html[OpenNMS deployment] for help on getting started with OpenNMS.)
 
 We also recommend setting up link:https://grafana.com/plugins/opennms-helm-app[OpenNMS Helm] (requires 3.0.0 or greater) to view and manage the situations, as well as link:https://www.elastic.co/products/elasticsearch[Elasticsearch] (requires 6.3.0 or greater) to maintain event and alarm history for training purposes.
```
```suggestion
We also recommend setting up https://grafana.com/plugins/opennms-helm-app[OpenNMS Helm] (3.0.0 or greater is required) to view and manage situations, as well as https://www.elastic.co/products/elasticsearch[Elasticsearch] (6.3.0 or greater is required) to maintain event and alarm history for training purposes.
```

emarsh1995 commented on docs/modules/install/pages/requirements.adoc:
```diff
@@ -1,8 +1,7 @@
 
 We also recommend setting up link:https://grafana.com/plugins/opennms-helm-app[OpenNMS Helm] (requires 3.0.0 or greater) to view and manage the situations, as well as link:https://www.elastic.co/products/elasticsearch[Elasticsearch] (requires 6.3.0 or greater) to maintain event and alarm history for training purposes.
-The installation guide will assume that these are also set up.
+(See the https://docs.opennms.com/helm/latest/index.html[HELM documentation] and https://docs.opennms.com/horizon/31/operation/deep-dive/elasticsearch/introduction.html[Elasticsearch integration] for more information.)
```
```suggestion
See the https://docs.opennms.com/helm/latest/index.html[HELM documentation] and https://docs.opennms.com/horizon/latest/operation/deep-dive/elasticsearch/introduction.html[Elasticsearch integration] documentation for more information.
```

emarsh1995 commented on docs/modules/reference/pages/model.adoc:
```diff
@@ -1,17 +1,17 @@
 = Model
```
```suggestion

```

emarsh1995 commented on docs/modules/reference/pages/model.adoc:
```diff
@@ -1,17 +1,17 @@
 = Model
-:imagesdir: ../assets/images 
+:imagesdir: ../assets/images
```
```suggestion
:imagesdir: ../assets/images
= Model
```

emarsh1995 commented on docs/modules/reference/pages/model.adoc:
```diff
@@ -1,17 +1,17 @@
 = Model
-:imagesdir: ../assets/images 
+:imagesdir: ../assets/images
 
```
It would be nice to add a page description here, like we have on other pages. As it is, I'm not sure what "model" refers to.

emarsh1995 commented on docs/modules/reference/pages/model.adoc:
```diff
@@ -1,17 +1,17 @@
-The following types are supported using the built in Inventory Scripts:
+The following types are supported using the built-in inventory scripts:
 
 === Node (node)
```
Maybe we could use the `Node:: <definition>` syntax for this section; it might look cleaner than having many short sections.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - updated the syntax for code samples.

Bonrob2 committed:
Apply suggestions from code review

Co-authored-by: Emily <107147744+emarsh1995@users.noreply.github.com>

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - updated based on reviewer feedback. Shortened some of the procedures as well, so as not to have so many code blocks of only one line.

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,136 @@
+
+You must complete the xref:install:pre_install.adoc[pre-installation tasks] before proceeding with the installation.
+
+Make sure that you install the package repository on all systems that are running Sentinel and OpenNMS.
```
This might not belong in the "Before you begin" section... Perhaps it should be moved to the section on installing the packages?

emarsh1995 commented on docs/modules/admin/pages/distributed_install.adoc:
```diff
@@ -0,0 +1,136 @@
+Complete the following steps in the OpenNMS Karaf shell.
+They configure your OpenNMS instance in preparation for a distributed installation of ALEC.
+
+.Configure the Kafka Producer to point to your Kafka broker:
```
```suggestion
.Configure Kafka Producer to point to your Kafka broker:
```

emarsh1995 commented on docs/modules/admin/pages/verifying.adoc:
```diff
@@ -0,0 +1,71 @@
+= Maintenance Tasks
+
+This section describes maintenance tasks that you may want to perform.
+They apply to basic and distributed deployments.
```
```suggestion
They apply to both basic and distributed installations.
```

emarsh1995 reviewed (changes requested):
Just a few nits. :)

Bonrob2 committed:
Apply suggestions from code review

Co-authored-by: Emily <107147744+emarsh1995@users.noreply.github.com>

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section: updated based on reviewer feedback. Moved location of before you begin content in the distributed install chapter.

Bonrob2 committed:
ALEC-227: ALEC docs: update installation section

ALEC-227: ALEC docs: update installation section - updated the health-check message based on reviewer feedback. Note that feedback did not appear in the PR, but was sent via chat.

Bonrob2 committed:
Merge branch 'develop' into jira/ALEC-227

Bonrob2 commented:
@BenjaminJ , I have updated based on the screenshot of your review that you shared in the chat. It did not show up on GitHub. Please verify the update is correct. I'm confused as to why there would be a success message, when Elasticsearch is not configured.

emarsh1995 reviewed (approved):
LGTM! Thank you! :D
